ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,273 @@
+        sbref_data = vol_reshape_and_mask(SBREFOrig, volumeValid);
+        
+        wm_data(wm_data ~= 0) = 1;
+        csf_data(csf_data ~= 0) = 1;
+        noisy_data=(sum(Volume_data,2)./sbref_data)>prctile(sum(Volume_data,2)./sbref_data,87.5);
+        edge_data=wmparc_data==0;
+        Volume_data=Volume_data./std(Volume_data(:));
+        CIFTI.cdata=CIFTI.cdata./std(CIFTI.cdata(:));
+        
+        nongray_data(nongray_data ~= 0) =1;
+
+        NonGrey=nongray_data;
+        NonGreyPlusEdge=NonGrey+edge_data;
+        Grey=(NonGreyPlusEdge-1)*-1;
+        VolSmoothROIs_data=wmparc_data.*0;
+        VolSmoothROIs_data(NonGrey==1)=1;
+        VolSmoothROIs_data(Grey==1)=2;
+        VolSmoothROIs_data(edge_data==1)=3;
+        VOLSMOOTHROIS=WMPARC.*0;
+        VOLSMOOTHROIS(std(VOLUME,[],2)>0)=VolSmoothROIs_data;
+        VOLSMOOTHROIS=reshape(VOLSMOOTHROIS,size(SBREFOrig,1),size(SBREFOrig,2),size(SBREFOrig,3),size(SBREFOrig,4));
+        save_avw(VOLSMOOTHROIS,[SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz'],'f',[1.6 1.6 1.6 1]);
+        system(['fslcpgeom ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_SBRef.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz -d']);
+        
+        SmoothingSigma=FinalSpatialSmoothingFWHM/(2*sqrt(2*log(2)));
+        system(['wb_command -volume-label-import ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz "" ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz']);
+        system(['wb_command -volume-parcel-smoothing ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.nii.gz ' SubjFolderlist '/MNINonLinear/ROIs/VolumeSmoothROIs.' Resolution '.nii.gz ' num2str(SmoothingSigma) ' ' SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+
+        VOLUMESMOOTH=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC_FWHM' num2str(FinalSpatialSmoothingFWHM) '.nii.gz']);
+        VOLUMESMOOTH=reshape(VOLUMESMOOTH,size(VOLUMESMOOTH,1)*size(VOLUMESMOOTH,2)*size(VOLUMESMOOTH,3),size(VOLUMESMOOTH,4));
+        VolumeSmooth_data=VOLUMESMOOTH(std(VOLUMESMOOTH,[],2)>0,:);
+        VolumeSmooth_data=VolumeSmooth_data./std(Volume_data(:));
+        CIFTIMSMAllSmooth.cdata=CIFTIMSMAllSmooth.cdata./std(CIFTIMSMAll.cdata(:));
+        
+        % structure idx
+        CIFTI_LENGTH=length(CIFTI.cdata);
+        CORTEX_start_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_LEFT_end_idx=max(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_LEFT').ciftilist);
+        CORTEX_RIGHT_end_idx=min(cifti_diminfo_dense_get_surface_info(CIFTI.diminfo{1}, 'CORTEX_RIGHT').ciftilist);
```

Comment:
So, `CORTEX_RIGHT_end_idx` is actually the first right index, so that is one mystery solved...the CEREBELLUM typos still look like a actual errors affecting the results.

Reply:
This typo is only included after this commit so it won't affect the weights in the commit.